src/themes/teams/components/Icon/iconStyles.ts

```diff
@@ -53,21 +58,23 @@ const getXSpacingStyles = (xSpacing: IconXSpacing, horizontalSpace: string): ICS
   }
 }
 
-const getBorderedStyles = (isFontBased, circular, borderColor, color): ICSSInJSStyle => {
+const getBorderedStyles = (circular, borderColor, color, boxShadowFallbackColor): ICSSInJSStyle => {
```

Comment:
seems that both `borderColor`, `color` and `boxShadowFallbackColor` are the arguments that should provide `boxShadow` color to the calculated styles object - this is the only responsibility for each of them. Thus, I would suggest to simplify a contract of this function:

```js
const getBorderedStyles = (circular,  boxShadowColor) => {
  ... 
  return {
    ...
    boxShadow: `... ${boxShadowColor}`
  }
}
```

and do all the necessary reductions on the client side:

```js
getBorderedStyles(circular, v.borderColor || v.color || theme.siteVariables.bodyColor)
```

This will make code more expressive, as well as a contract for this function to be much cleaner (and we won't need to introduce additional parameters if another fallback value will come into play :).

Reply:
Makes sense. I will make this update.